lib/portage/sync/modules/git/git.py

```diff
@@ -228,6 +228,24 @@ def update(self):
         if not self.verify_head(revision="refs/remotes/%s" % remote_branch):
             return (1, False)
 
+        # Clean up the repo before trying to sync to upstream's
+        clean_cmd = [self.bin_command, "clean", "--force", "-d", "-x"]
```

Comment:
Ceiling dir?

Reply:
Is that something to handle in this PR? There's other git invocations than this one that would need it, right?

Reply:
I think this is the only place we call git clean which is destructive for unknown files.